src/line.ts

```diff
@@ -1,83 +1,92 @@
 import * as d3 from 'd3';
-import { DateTime } from 'luxon'
-import { generateCube } from './datagen/generation';
-import { categories, measures } from './data_config';
-import { betweenDates } from './datagen/filters';
+import {DateTime} from 'luxon';
+import {generateCube} from './datagen/generation';
+import {categories, measures} from './data_config';
+import {betweenDates} from './datagen/filters';
 
-interface Data {
+interface Datum {
   date: Date;
   value: number;
 }
 
 const cube = generateCube(categories, measures);
 
-function getData(): Data[] {
+function getData(): Datum[] {
   const endDate = DateTime.local();
-  const startDate = endDate.minus({ day: 30 });
+  const startDate = endDate.minus({day: 30});
 
   const data = cube.getDataFor(
-    ['nthDay'], 
-    ['activeUsers'], 
-    [betweenDates(
-        new Date(startDate.toMillis()), 
-        new Date(endDate.toMillis()))
-        ]
-        );
-  return data.map((datum) => ({
-      date: startDate.plus({ days: datum.categories.get('nthDay') }),
-      value: datum.values.get('activeUsers')!,
+    ['nthDay'],
+    ['activeUsers'],
+    [betweenDates(startDate.toJSDate(), endDate.toJSDate())]
+  );
+  return data.map(datum => ({
+    date: startDate
+      .plus({days: datum.categories.get('nthDay') as number})
```

Comment:
Generally better to use the `!` operator here than a cast, because it makes it more clear that you're only saying the undefined value is guaranteed not to be there.

Reply:
Hi @danielyule, `days` requires to be of `number | undefined` type while `datum.categories.get('nthDay')` is of `string | number | undefined`, so I had to explicitly cast it to `number`.